modified:   app.js

DamianDawidowski opened the pull request:
new file:   routes/authorization.js
 	new file:   service/authorization.js
 	new file:   service/user.js

DamianDawidowski committed:
modified:   app.js
 	new file:   routes/authorization.js
 	new file:   service/authorization.js
 	new file:   service/user.js

RadziuMM reviewed (changes requested):
1. Jak już kopiujesz z kogoś to musisz odpalić ten kod nawet po to by dowiedzieć się że masz niezainstalowane zależności (paczki za pomocą npm) bo teraz widać że nawet tego nie odpaliłeś....
2. W kilku miejscach masz exporty obiektów a importujesz je jak by była tam tylko 1 funkcja np
```
module.exports = { createUser, ....., router }
const authRouter = require("./routes/authorization"); // źle
const { router: authRouter } = require("./routes/authorization"); // dobrze
```
3. Kod nie jest sformatowany i się rozjeżdża.
4. W konsoli jeszcze wyskakuje kilka błędów które też trzeba naprawić

DamianDawidowski committed:
modified:   app.js
 	modified:   package-lock.json
 	modified:   package.json
 	modified:   routes/authorization.js
 	modified:   routes/routes.js
 	modified:   service/authorization.js
 	new file:   service/functions.js

DamianDawidowski committed:
modified:   routes/authorization.js
 	modified:   routes/routes.js
 	modified:   service/authorization.js
 	modified:   service/functions.js
 	modified:   service/user.js

DamianDawidowski committed:
modified:   app.js

DamianDawidowski commented:
Zainstalowałem zależności, uporządkowałem kod i przetestowałem funkcje w Postman, wszystkie działają.

RadziuMM commented on routes/routes.js:
```diff
@@ -1,8 +1,9 @@
  
-// const router = express.Router();
+router.use("/api/contacts", contactsRouter);
+router.use("/users", authRouter);
```
router.use("/api/users", authRouter);

RadziuMM commented on service/authorization.js:
```diff
@@ -0,0 +1,44 @@
+  }  
+};
+
+const auth = async (req, res, next) => {
```
Ogólnie to działa i nawet dobrze, ale chcemy zastosować tutaj strategię oraz użyć jedngo z bardziej znanych modeli tej strategii czyli bearer tokenu (Tak w skrócie jest to prawie to samo tylko że z przodu jest słówko Bearer)

Pokazane jak to zrobić masz na lms w materiałach Moduł 4 lekcja 7-8 autoryzacja jwt (tam praktycznie wystarczy to skopiować, jedyne co mususz dodać to sprawdzanie czy token w req będzie taki sam jak w w db bo tego przykład nie obejmuje)

DamianDawidowski committed:
modified:   app.js
 	modified:   package-lock.json
 	modified:   package.json
 	modified:   service/authorization.js
 	new file:   service/config-passport.js

DamianDawidowski commented:
Dodałem autoryzację jwt, testowałem i działa w Postman.
